### DISCUSSION TITLE
Memory safety with out-of-range sprites

### DISCUSSION THREAD

**Discussion by motoani** [Bodmer/TFT_eSPI#2325] [Q&A - Sprites] [answered]
I'm finding this library a massive asset and time saver but am wondering if I'm over complicating. My quick scan of the Sprite.cpp suggests that it's not safe to push a sprite to a sprite with negative position which would be useful to allow a sprite to pass beyond (0,0) and move off screen smoothly. There is a cropped sprite  to TFT method, but not sprite to sprite.  Do I have to make a set of narrow sprites to pass off the screen edge?

**Comment by Bodmer** [comment, marked as answer]
The library automatically handles the clipping (via the PI_CLIP macro), so this is safe.

**Reply by motoani** [reply]
Thank you, I've revisited the code and noted this macro and role in the cpp. Also implemented out of range pushes and all seems well. Pasting off the field of view with a changing offset is a far quicker way to move a background sprite than actually scrolling it (by 10ms per frame on ESP32 320x170).
